### PR TITLE
feat: enable connectionless proof

### PR DESCRIPTION
**Pull request by morrieinmaas**
* create invite works
* accept invite works
* both OOB
* BUT: no proof request is going through

Signed-off-by: Moriarty <moritz@animo.id>

### CODE REVIEW

**Comment by morrieinmaas** [comment]
closes #265

File: app/generic/verifier/verifier.py [hunk 1]
```diff
@@ -239,9 +238,12 @@ async def accept_proof_request(
     try:
         prover = __get_verifier_by_version(presentation.proof_id)
 
-        await assert_valid_prover(
-            aries_controller=aries_controller, prover=prover, presentation=presentation
-        )
+        if not presentation.connectionless:
```
Comment (TimoGlastra) [inline app/generic/verifier/verifier.py:241]: We can check whether the proof is connectionless by first retrieving it from the ACA-Py instance. This prevents the user needing to provide info that is already availalbe to use and also removes a way for inconsistent state: the user sets connectionless: false, but the exchange is actually connectionless

File: app/generic/verifier/models.py [hunk 2]
```diff
@@ -27,6 +27,7 @@ class CreateProofRequest(BaseModel):
 
 class AcceptProofRequest(BaseModel):
     proof_id: str
+    connectionless: Optional[bool] = False
```
Comment (TimoGlastra) [inline app/generic/verifier/models.py:30]:
```suggestion
```

File: app/generic/verifier/verifier.py [hunk 3]
```diff
@@ -177,9 +176,9 @@ async def send_proof_request(
     try:
         prover = __get_verifier_by_version(proof_request.protocol_version)
 
-        await assert_valid_verifier(
-            aries_controller=aries_controller, proof_request=proof_request
-        )
+        # await assert_valid_verifier(
+        #     aries_controller=aries_controller, proof_request=proof_request
+        # )
```
Comment (TimoGlastra) [inline app/generic/verifier/verifier.py:181]: I think we only want to disable this for connectionless proof requests

```suggestion
        await assert_valid_verifier(
            aries_controller=aries_controller, proof_request=proof_request
        )
```

File: app/generic/verifier/verifier.py [hunk 4]
```diff
@@ -246,9 +238,12 @@ async def accept_proof_request(
     try:
         prover = __get_verifier_by_version(presentation.proof_id)
 
-        await assert_valid_prover(
-            aries_controller=aries_controller, prover=prover, presentation=presentation
-        )
+        if not presentation.connectionless:
+            await assert_valid_prover(
```
Comment (TimoGlastra) [inline app/generic/verifier/verifier.py:242]: Althoug we can't check the connection, we _could_ check whether the proof request is valid in terms of the schema check. the assert_valid_prover does two checks

File: app/generic/verifier/verifier.py (hunk 1 above)
Comment (morrieinmaas) [reply app/generic/verifier/verifier.py:241]: ah alright nice didnt know that - will implement that then

File: app/generic/verifier/verifier.py (hunk 3 above)
Comment (morrieinmaas) [reply app/generic/verifier/verifier.py:181]: yes

**Review by TimoGlastra — APPROVED** [review]

**Pull request merged**